File: Programming Basics/C++/Nested loops/TrainTheTrainers/TrainTheTrainers.cpp
#include <iostream>
#include <string>
using namespace std;

int main()
{
    int juries;
    cin >> juries;

    double presentationScoresSum = 0;
    int presentationsCount = 0;

    cout.setf(ios::fixed);
    cout.precision(2);

    string presentationName;
    cin >> ws;
    getline(cin, presentationName);
    while (presentationName != "Finish")
    {
        double currentScoresSum = 0;
        for (int i = 0; i < juries; i++)
        {
            double juryScore;
            cin >> juryScore;
            currentScoresSum += juryScore;
        }

        double grade = currentScoresSum / juries;
        cout << presentationName << " - " << grade << "." << endl;

        presentationScoresSum += grade;
        presentationsCount++;

        cin >> ws;
        getline(cin, presentationName);
    }

    double averageAssessment = presentationScoresSum / presentationsCount;
    cout << "Student's final assessment is " << averageAssessment << "." << endl;
}

File: Programming Basics/JavaScript/Nested loops/Exercises/train-the-trainers.js
function solve(input) {
    const juries = Number(input[0]);

    let presentationScoresSum = 0;
    let presentationsCount = 0;

    let index = 1;
    let presentationName = input[index];
    while (presentationName !== "Finish") {
        let currentScoresSum = 0;
        for (let i = 0; i < juries; i++) {
            const juryScore = Number(input[++index]);
            currentScoresSum += juryScore;
        }

        const grade = currentScoresSum / juries;
        console.log(`${presentationName} - ${grade.toFixed(2)}.`);

        presentationScoresSum += grade;
        presentationsCount++;

        presentationName = input[++index];
    }

    const averageAssessment = presentationScoresSum / presentationsCount;
    console.log(`Student's final assessment is ${averageAssessment.toFixed(2)}.`);
}
